### PR TITLE
os/wqueue : Refactor work queue

### PR DESCRIPTION
**Pull request by YiHyunJin**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5d2047041b9e8d904ec240b6fd971fc8d54f520f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5d2047041b9e8d904ec240b6fd971fc8d54f520f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0b6f809b1fdd4fdf8ba3e5f14f58804c640f51d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0b6f809b1fdd4fdf8ba3e5f14f58804c640f51d2] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
And please split the changes for the second commit.
1. remove unnecessary variable, delay
2. remove the value of wqueue configuration removed from defconfig

**Comment by seinfra** [comment]
Target : [8524fa98b639e9ec807f1d2bc78d47b6da040ac9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8524fa98b639e9ec807f1d2bc78d47b6da040ac9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [19cb9693a47e9611e29afcf3e75eb8f7d54600d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [19cb9693a47e9611e29afcf3e75eb8f7d54600d2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e8134fc9f628fc5393644c1a4dbdff5dd8876b1b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e8134fc9f628fc5393644c1a4dbdff5dd8876b1b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [96ef1194c3614266eb3fd1b1cfc931f05b6afc44] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [96ef1194c3614266eb3fd1b1cfc931f05b6afc44] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [05893c6f98600708392aaad2c8a81eb86d222ebd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [05893c6f98600708392aaad2c8a81eb86d222ebd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [240176cdbd6ba1f4270942ec09372f3e8b64fa42] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [240176cdbd6ba1f4270942ec09372f3e8b64fa42] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [deb5e5e7f2d5b09bc1ac22288e9627e36759dd2a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [deb5e5e7f2d5b09bc1ac22288e9627e36759dd2a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9ee5c204e220dbc0814047203876c02dcbab1f6b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9ee5c204e220dbc0814047203876c02dcbab1f6b] - Code Rule Check OK.

File: os/wqueue/kwqueue/kwork_signal.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#include <tinyara/config.h>
+
+#include <signal.h>
+#include <errno.h>
+
+#include <tinyara/wqueue.h>
+
+#include "wqueue.h"
+
+#ifdef CONFIG_SCHED_WORKQUEUE
```
[the remark this answers is not in the document]
Comment (YiHyunJin) [reply os/wqueue/kwqueue/kwork_signal.c]: Yes. Checking this in Make.defs does not add the c file, but I think it should be prevented if added the c file in another Make.defs. So I think you need a double check.

File: os/wqueue/uwqueue/uwork_signal.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#include <tinyara/config.h>
+
+#include <signal.h>
+#include <errno.h>
+
+#include <tinyara/wqueue.h>
+
+#include "wqueue.h"
+
+#if defined(CONFIG_SCHED_USRWORK) && !defined(__KERNEL__)
```
[the remark this answers is not in the document]
Comment (YiHyunJin) [reply os/wqueue/uwqueue/uwork_signal.c]: 1. Yes. Checking this in Make.defs does not add the c file, but I think it should be prevented if added the c file in another Make.defs. So I think you need a double check.
2. uwork should work when it's not a kernel build.

[the remark this answers is not in the document]
Comment (sunghan-chang) [reply os/wqueue/uwqueue/uwork_signal.c]: Let's talk this tomorrow again

**Comment by seinfra** [comment]
Target : [54b4ea395e6a6844c137c94e9313a847699a3a7f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [54b4ea395e6a6844c137c94e9313a847699a3a7f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d2752ee5e7966af7cd3336d1b3ee2cf3ac2c2a82] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d2752ee5e7966af7cd3336d1b3ee2cf3ac2c2a82] - Code Rule Check OK.

**Pull request merged**